### PR TITLE
simplify codes

### PR DESCRIPTION
**Pull request by minggo**
Reduce duplicated codes can reduce package size https://github.com/cocos/cocos-engine/issues/17715

### CODE REVIEW

File: cocos/ui/scroll-view.ts [hunk 1]
```diff
@@ -857,7 +787,11 @@ export class ScrollView extends ViewGroup {
      * ```
      */
     public scrollToPercentVertical (percent: number, timeInSecond?: number, attenuated?: boolean): void {
-        assignMoveDeltaOption(0, percent, false, true);
+        this._doScroll(0, percent, false, true, timeInSecond, attenuated);
+    }
+
+    private _doScroll (x: number, y: number, applyToHorizontal: boolean, applyToVertical: boolean, timeInSecond?: number, attenuated = true): void {
+        assignMoveDeltaOption(x, y, applyToHorizontal, applyToVertical);
```
Comment (qiuguohua) [inline cocos/ui/scroll-view.ts:794]: The function assignMoveDeltaOption can be removed?

Comment (minggo) [reply cocos/ui/scroll-view.ts:794]: No, it will update `_moveDeltaOptions`

Comment (qiuguohua) [reply cocos/ui/scroll-view.ts:794]: I mean to put the implementation of assignMoveDeltaOption directly here?

**Review by qiuguohua — APPROVED** [review]

File: cocos/ui/scroll-view.ts (hunk 1 above)
Comment (minggo) [reply cocos/ui/scroll-view.ts:794]: It is better not. Current implementation make it more clearly.

**Pull request merged**